org.lflang/src/org/lflang/generator/c/CGenerator.java

```diff
@@ -609,21 +585,32 @@ public void doGenerate(Resource resource, LFGeneratorContext context) {
                     dockerGenerator.fromData(lfModuleName, federate.name, fileConfig));
             }
             // If cmake is requested, generate the CMakeLists.txt
-            var cmakeFile = fileConfig.getSrcGenPath() + File.separator + "CMakeLists.txt";
-            var cmakeCode = cmakeGenerator.generateCMakeCode(
-                List.of(cFilename),
-                lfModuleName,
-                errorReporter,
-                CCppMode,
-                mainDef != null,
-                cMakeExtras,
-                targetConfig
-            );
-            try {
-                cmakeCode.writeToFile(cmakeFile);
-            } catch (IOException e) {
-                //noinspection ThrowableNotThrown,ResultOfMethodCallIgnored
-                Exceptions.sneakyThrow(e);
+            if (targetConfig.platformOptions.platform != Platform.ARDUINO){
+                var cmakeFile = fileConfig.getSrcGenPath() + File.separator + "CMakeLists.txt";
+                var cmakeCode = cmakeGenerator.generateCMakeCode(
+                    List.of(cFilename),
+                    lfModuleName,
+                    errorReporter,
+                    CCppMode,
+                    mainDef != null,
+                    cMakeExtras,
+                    targetConfig
+                );
+                try {
+                    cmakeCode.writeToFile(cmakeFile);
+                } catch (IOException e) {
+                    //noinspection ThrowableNotThrown,ResultOfMethodCallIgnored
+                    Exceptions.sneakyThrow(e);
+                }
+            } else {
+                try {
+                    Files.deleteIfExists(fileConfig.getSrcGenPath().resolve("src/lib/util.c"));
```

Comment:
Why is this happening, and why is `src/lib/util.c` mentioned separately?

Reply:
util.c is a conflicting filename, I need to remove it here before doing relative path changing.

Reply:
This contains deprecated functions that are no longer in use as well.

Reply:
What's the conflict?

Reply:
The way I change relative paths is to create a map between the filename and its Path. There is util.c inside of lib and util.c inside of core/utils, which will create a conflict. We don't use util.c afaik since the functions are out of date, so I opted to take it out before relativizing paths.

Reply:
Oh, I see. I wasn't aware that we had two `util.c` files... Your solution sounds reasonable but we probably shouldn't have these conflicting files to begin with. A related issue, https://github.com/lf-lang/reactor-c/issues/111, came up recently. Maybe @erlingrj or @petervdonovan want to weigh in on this.

Reply:
We should change that. The `util.c` that you are talking about implements deprecated API functions. We could rename it to `depreacted.c`? Or something similar.

Reply:
Please update `reactor-c` to `2553fc84899128a8ba44f3173c287bb986578661` and remove the special case of `util.c`. After that, this conversation can be marked "resolved."